Tests/AppStoreConnectClientTests/RateLimitMiddlewareTests.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  You should have received a copy of the GNU Affero General Public License
+//  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//
+//  RateLimitMiddlewareTests 2.swift
+//
+//
+//  Created by Mykola Vasyk on 15.05.2024.
+//
+
+import XCTest
+import HTTPTypes
+import OpenAPIRuntime
+@testable import AppStoreConnectClient
+
+class RateLimitMiddlewareTests: XCTestCase {
+    func testExtractHeaderValueSuccess() throws {
+        let sut = RateLimitMiddleware()
+        var fields = HTTPFields()
+        let unwrap = try XCTUnwrap(HTTPField.Name("x-rate-limit"))
+        fields[unwrap] = "user-hour-lim:1111; user-hour-rem:222"
+        let mockResponse = HTTPTypes.HTTPResponse(
+            status: .accepted,
+            headerFields: HTTPFields(fields[fields: unwrap])
+        )
+        XCTAssertNoThrow(try sut.extractHeaderValue(from: mockResponse, for: "x-rate-limit"))
+    }
+    
+    func testExtractHeaderValueInvalidSearchingDataHeader() throws {
+        let sut = RateLimitMiddleware()
+        var fields = HTTPFields()
+        let unwrap = try XCTUnwrap(HTTPField.Name("x-limit"))
+        fields[unwrap] = "user-hour-lim:1111; user-hour-rem:222"
+        let mockResponse = HTTPTypes.HTTPResponse(
+            status: .accepted,
+            headerFields: HTTPFields(fields[fields: unwrap])
+        )
+        do {
+            _ = try sut.extractHeaderValue(from: mockResponse, for: "x-rate-limit")
+            XCTFail("Expected error not thrown")
+        } catch RateLimitError.invalidSearchingData(header: "x-rate-limit") {
+        } catch {
+            XCTFail("Unexpected error: \(error)")
+        }
+    }
+    
+    func testExtractHeaderValueRateLimitExceeded() throws {
+        let limit = 1111
+        let sut = RateLimitMiddleware()
+        var fields = HTTPFields()
+        let unwrap = try XCTUnwrap(HTTPField.Name("x-rate-limit"))
+        fields[unwrap] = "user-hour-lim:1111; user-hour-rem: 0"
+        let mockResponse = HTTPTypes.HTTPResponse(
+            status: .accepted,
+            headerFields: HTTPFields(fields[fields: unwrap])
+        )
+        do {
+            _ = try sut.extractHeaderValue(from: mockResponse, for: "x-rate-limit")
+            XCTFail("Expected error not thrown")
+        } catch RateLimitError.rateLimitExceeded(remaining: 0, from: limit) {
+        } catch {
+            XCTFail("Unexpected error: \(error)")
+        }
+    }
+    
+    func testExtractHeaderInvalidExpectedValues() throws {
+        let sut = RateLimitMiddleware()
+        var fields = HTTPFields()
+        let unwrap = try XCTUnwrap(HTTPField.Name("x-rate-limit"))
+        fields[unwrap] = "user-hour-lim: user-hour-rem: user-hour-rem:"
+        let mockResponse = HTTPTypes.HTTPResponse(
+            status: .accepted,
+            headerFields: HTTPFields(fields[fields: unwrap])
+        )
+        do {
+            _ = try sut.extractHeaderValue(from: mockResponse, for: "x-rate-limit")
```

Comment:
extractHeaderValue must be private and therefore this test is not correct

Reply:
Refactored.